### PR TITLE
Restrict Dropdown

### PR DESCRIPTION
**Pull request by jorgemoya**
Added new `options` prop that will restrict the usage of the items. The `onClick` for each item with return the `item` itself. No need to use `data-value` anymore.

Now also accepts `icons` as a DropdownItem prop.

Screen Shot 2019-10-15 at 8 44 11 PM

Wasn't sure how to document the new `DropdownItem` and `DropdownLinkItem`. Feedback pls 🙏 .

docs2

### CODE REVIEW

**Comment by deini** [comment]
Should all the options use a pointer cursor? 🤔

File: packages/big-design/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -58,46 +75,80 @@ export class Dropdown extends React.PureComponent<DropdownProps, DropdownState>
...
-            ref,
-            role: 'option',
-          }) as React.LiHTMLAttributes<HTMLLIElement>;
-        default:
-          return;
-      }
-    });
+  private renderIcon(icon: React.ReactNode, isHighlighted: boolean, option: DropdownItem | DropdownLinkItem) {
+    return (
+      React.isValidElement(icon) &&
```
Comment (deini) [inline packages/big-design/src/components/Dropdown/Dropdown.tsx]: Can we add a check to make sure the type is icon?

File: packages/big-design/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -58,46 +75,80 @@ export class Dropdown extends React.PureComponent<DropdownProps, DropdownState>
...
+
+        return (
+          <ListItem
+            {...rest}
+            actionType={actionType}
+            data-highlighted={isHighlighted}
+            id={id}
+            key={index}
+            onClick={() => this.handleOnItemClick(option)}
+            onFocus={this.handleOnItemFocus}
```
Comment (deini) [inline packages/big-design/src/components/Dropdown/Dropdown.tsx]: Should `onFocus` and `onMouseOver` call the same fn?

File: packages/big-design/src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -4,26 +4,42 @@ import { Manager, Reference, RefHandler } from 'react-popper';
...
 
-export class Dropdown extends React.PureComponent<DropdownProps, DropdownState> {
-  static Item = ListItem;
+export interface DropdownLinkItem extends BaseItem {
+  target?: HTMLAnchorElement['target'];
+  type: 'link';
+  url: string;
+}
 
+interface BaseItem extends Omit<ListItemProps, 'children' | 'content' | 'onClick' | 'value'> {
```
Comment (deini) [inline packages/big-design/src/components/Dropdown/Dropdown.tsx]: Can you move the `BaseItem` definition before `DropdowItem` and `DropdownLinkItem` :pray:

**Comment by jorgemoya** [comment]
> Should all the options use a pointer cursor? 🤔

Let me consult with design.

File: packages/big-design/src/components/Dropdown/spec.tsx [hunk 4]
```diff
@@ -6,13 +7,18 @@ import { Button } from '../Button';
...
+const onClick = jest.fn();
+
 const DropdownMock = (
-  <Dropdown trigger={<Button>Button</Button>}>
-    <Dropdown.Item value={0}>Option</Dropdown.Item>
-    <Dropdown.Item value={1}>Option</Dropdown.Item>
-    <Dropdown.Item value={2}>Option</Dropdown.Item>
-    <Dropdown.Item value={3}>Option</Dropdown.Item>
-  </Dropdown>
+  <Dropdown
```
Comment (deini) [inline packages/big-design/src/components/Dropdown/spec.tsx]: 🍹

```jsx
<Dropdown
  options={[
    { content: 'Option', type: 'string', value: '0' },
    { content: 'Option', type: 'string', value: '1', onClick },
    { content: 'Option', type: 'string', value: '2', actionType: 'destructive' },
    { content: 'Option', type: 'string', value: '3', icon: <CheckCircleIcon /> },
  ]}
  trigger={<Button>Button</Button>}
/>
```

File: packages/big-design/src/components/Dropdown/spec.tsx [hunk 5]
```diff
@@ -257,3 +238,32 @@ test('dropdown items should be highlighted when moused over', () => {
...
+    <Dropdown
+      onClick={onClick}
+      options={[
+        { content: 'Option', url: '#', type: 'link' },
+        { content: 'Option', url: '#', type: 'link' },
+        { content: 'Option', url: '#', type: 'link' },
+        { content: 'Option', url: '#', type: 'link' },
+      ]}
+      trigger={<Button>Button</Button>}
+    ></Dropdown>,
```
Comment (deini) [inline packages/big-design/src/components/Dropdown/spec.tsx]: Same as above.

File: packages/big-design/src/components/Dropdown/Dropdown.tsx [hunk 6]
```diff
@@ -58,46 +52,79 @@ export class Dropdown extends React.PureComponent<DropdownProps, DropdownState>
...
+          >
+            {option.type === 'link' && !option.disabled ? (
+              <Link href={option.url} target={option.target}>
+                <Flex alignItems="center">
+                  {icon && <FlexItem paddingRight="small">{this.renderIcon(option, isHighlighted)}</FlexItem>}
+                  {content}
+                </Flex>
+              </Link>
+            ) : (
+              <Flex alignItems="center">
```
Comment (chanceaclark) [inline packages/big-design/src/components/Dropdown/Dropdown.tsx]: Same as above

**Pull request merged**